src/main/java/uk/gov/hmcts/reform/prl/handlers/TransferToAnotherCourtEventHandler.java

```diff
@@ -0,0 +1,158 @@
+package uk.gov.hmcts.reform.prl.handlers;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.prl.enums.LanguagePreference;
+import uk.gov.hmcts.reform.prl.enums.YesNoDontKnow;
+import uk.gov.hmcts.reform.prl.events.TransferToAnotherCourtEvent;
+import uk.gov.hmcts.reform.prl.models.Element;
+import uk.gov.hmcts.reform.prl.models.complextypes.PartyDetails;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.models.email.EmailTemplateNames;
+import uk.gov.hmcts.reform.prl.services.EmailService;
+import uk.gov.hmcts.reform.prl.services.transfercase.TransferCaseContentProvider;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.prl.enums.YesOrNo.Yes;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TransferToAnotherCourtEventHandler {
+    private final EmailService emailService;
+    private final TransferCaseContentProvider transferCaseContentProvider;
+
+    @EventListener(condition = "#event.typeOfEvent eq 'Transfer to another court'")
+    public void notifyAllParties(final TransferToAnotherCourtEvent event) {
+        CaseData caseData = event.getCaseData();
+
+        sendEmailToApplicantSolicitor(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_APPLICANT);
+
+        sendEmailToApplicant(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_APPLICANT);
+
+        sendEmailToRespondentSolicitor(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_RESPONDENT);
+
+        sendEmailToRespondents(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_RESPONDENT);
+
+        sendEmailToOtherParties(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_OTHER_PARTIES);
+    }
+
+    private void sendEmailToRespondents(CaseData caseData, EmailTemplateNames emailTemplateNames) {
+        Map<String, String> respondentEmail = caseData.getRespondents().stream()
+            .map(Element::getValue)
+            .filter(respondent -> !CaseUtils.hasLegalRepresentation(respondent)
+                && Yes.equals(respondent.getCanYouProvideEmailAddress()))
+            .collect(Collectors.toMap(
+                PartyDetails::getEmail,
+                party -> party.getFirstName() + " " + party.getLastName(),
```

Comment:
use the method from Party object

Reply:
used common method instead of individual method to get values